### DISCUSSION TITLE
Any method for compact flaps control?

### DISCUSSION THREAD

**Discussion by beckerdo** [nguyenquyhy/Flight-Tracker-StreamDeck#204] [General]
Like many people, I am using a Generic Toggle Button to show my flaps position using the display value FLAPS_HANDLE_INDEX.

Does anyone know if it is possible to access toggle events such as FLAPS_UP/FLAPS_DOWN or FLAPS_INCR/FLAPS_DECR using the same button. Perhaps a single press for FLAPS_INCR and a double press for FLAPS_DECR.

I'd like to keep my Stream deck compact, and not dedicate additional buttons for moving flaps up and down.

**Comment by beckerdo** [comment]
FYI, **version 0.30** with support for "**Add Hold Event for Generic Toggle Button**" worked pretty well for my request. I was able to map FLAPS_INCR to the Toggle event, FLAPS_DECR to the Hold Event, and FLAPS_HANDLE_INDEX to the Display Value. This allows you with a button press to lower the flaps a notch, and with a long hold to move them up.

There is a minor gotcha here. I think when you hold you see a toggle event, followed by a hold event. Be sure you hold long enough to undo the toggle and follow it up with a hold.

Thank you!

**Reply by nguyenquyhy** [reply]
Yep I'll miss that out but it accidentally create a nice feature when you can activate 2 events with 1 click. I think I'll add in a checkbox or something to allow switching between trigging Toggle event when holding or not.
